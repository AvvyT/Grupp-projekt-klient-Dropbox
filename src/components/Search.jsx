import React, { useState } from "react";
import { dbx } from "./functions";
import styles from "./css/main.module.css";

// Visa sökresultat istället för huvudinnehållet
// Om en användare klickar på en katalog ska applikationen navigera till den katalogen
// Om en användare klickar på en fil ska filen laddas ner


const Search = () => {
  const [searchWord, updateSearchWord] = useState('');

  // En användare ska kunna söka efter filer och kataloger
  const searchFileOrFolder = () => {
    console.log(searchWord);

    // path in the user's Dropbox to search. Should probably be a folder

    // let results = [];
    let path = '';
    // query is For file name and folder searching 
    // starting index within the search results (used for paging)
    //	max_results => The maximum number of search results to return.
    dbx.filesSearch({
      path: path, query: searchWord, start: 0, max_results: 10, mode: { '.tag': 'filename_and_content' }
    })
      .then((response) => {
        console.log(response);
      })
      .catch((error) => {
        console.error(error);
      });

  }

  const handleSubmit = (e) => {
    e.preventDefault();

    searchFileOrFolder();
    updateSearchWord('');
  }

  return (
    <form className={styles.fromStyle} onSubmit={handleSubmit}>
      <input placeholder="search" className={styles.inputStyle} value={searchWord}
        onChange={(e) => {
          updateSearchWord(e.target.value);
          console.log(searchWord.length);
          
        }} />
      <input type='submit' className={styles.uploadButtonStyle} value='Search' />
    </form>
  );
}

export default Search;
