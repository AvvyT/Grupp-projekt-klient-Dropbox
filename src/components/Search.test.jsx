import React from 'react';
import ReactDOM from 'react-dom';
import Search from './Search.jsx';

/** ---I combine two testing approaches: Snapshot Testing and Component logic testing---**/
/**
 * Snapshot Testing is a useful testing tool in case you want to be sure user interface hasn’t changed.
 **/
import { shallow, configure, mount } from 'enzyme';

import Adapter from "enzyme-adapter-react-16"
configure({ adapter: new Adapter() });

// skapa en mock fun.
const mockPrevent = jest.fn();

describe('Search', () => {
    it('should render correctly in "debug" mode', () => {
        const funComponent = shallow(<Search debug />);

        expect(funComponent).toMatchSnapshot();
        // toMatchSnapshot() => method creates Snapshot itself
    });

    // Basic Fun-component rendering
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Search />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    // Mock functions
    it('form gets submitted once', () => {
        const wrapper = shallow(<Search />);

        expect(wrapper.find('input').length).toBe(2)
        // preventDefault-metod: läggs mock-fun
        wrapper.find('form').simulate("submit", {
            preventDefault: mockPrevent
        })
        expect(mockPrevent.mock.calls.length).toBe(1)
    });

    /**---Snapshot Testing---**/
    // event-test interacting with a child component // mount-test
    it('should be possible to activate form-submit with Spacebar', () => {
        const funComponent = mount(<Search />);
        funComponent
            .find('form')
            .simulate('keydown', { keyCode: 32 });
        expect(funComponent).toMatchSnapshot();
        funComponent.unmount();
    });

    
    it('should render banner text correctly with given strings', () => {
        const strings = ['one', 'two'];
        const funComponent = shallow(<Search list={strings} />);
        expect(funComponent).toMatchSnapshot();
    });

});