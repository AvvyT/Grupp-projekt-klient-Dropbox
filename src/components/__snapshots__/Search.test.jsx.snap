// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search  render correctly  1`] = `ShallowWrapper {}`;

exports[`Search render correctly  1`] = `ShallowWrapper {}`;

exports[`Search should be possible to activate form-submit with Spacebar 1`] = `ReactWrapper {}`;

exports[`Search should render banner text correctly with given strings 1`] = `ShallowWrapper {}`;

exports[`Search should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`should be possible to activate form-submit with Spacebar 1`] = `ReactWrapper {}`;

exports[`should render banner text correctly with given strings 1`] = `ShallowWrapper {}`;
